docs/development/design/system-packages.rst

```diff
@@ -0,0 +1,100 @@
+Allow Installation of System Packages
+=====================================
+
+Currently we don't allow executing arbitrary commands in the build process.
+The more common use case is to install extra dependencies.
+
+.. contents::
+   :local:
+   :depth: 3
+
+Current status
+--------------
+
+There is a workaround when using Sphinx to run arbitrary commands,
+this is executing the commands inside the ``conf.py`` file.
+There isn't a workaround for MkDocs, but this problem is more common in Sphinx,
+since users need to install some extra dependencies in order to use autodoc or build Jupyter Notebooks.
+
+However, installation of some dependencies require root access,
+or are easier to install using ``apt``.
+Most of the CI services allow to use ``apt`` or execute any command with ``sudo``,
+so users are more familiar with that workflow.
+
+Some users use Conda instead of pip to install dependencies in order to avoid these problems,
+but not all pip users are familiar with Conda, or want to migrate to Conda just to use Read the Docs.
+
+Security concerns
+-----------------
+
+Builds are run in a Docker container,
+but the app controlling that container lives in the same server.
+Allowing to execute arbitrary commands with super user privileges may introduce some security issues.
+
+Exposing ``apt install``
+------------------------
+
+For the previous reasons we won't allow to execute arbitrary commands with root (yet),
+but instead allow only to install extra packages using ``apt``.
+
+We would expose this through the config file.
+Users will provide a list of packages to install, and under the hook we would run:
+
+- ``apt update -y``
+- ``apt install -y {packages}``
```

Comment:
We need to think how are we going to escape the `{packages}` here to avoid people executing random commands with root here.

Reply:
That's an implementation detail, but it's already done in the PR.